include/cetl/pf17/any.hpp

```diff
@@ -0,0 +1,51 @@
+/// @file
+/// Defines the C++17 `std::any` type and several related entities.
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_PF17_ANY_HPP_INCLUDED
```

Comment:
just omit this header for now. It's completely off the mark. What we actually want is to have something like this in the cetlpf.hpp

```
using bad_any_cast = std::bad_any_cast;

template <typename T> constexpr bool is_rtti_convertible = std::false_type::value;
using type_id = std::type_info;

// Forward declarations
template <std::size_t Footprint, bool Copyable, bool Movable, std::size_t Alignment> class any;

template <typename T> inline constexpr const type_id &type_id_value()
{
    return typeid(T);
}

template <typename T> using in_place_type_t = std::in_place_type_t<T>;

template <typename T> inline constexpr std::in_place_type_t<T> in_place_type;

template <std::size_t I> using in_place_index_t = std::in_place_index_t<I>;

template <std::size_t I> inline constexpr std::in_place_index_t<I> in_place_index;

// Forward declarations
template <std::size_t Footprint, bool Copyable, bool Movable, std::size_t Alignment> class any;

```

Reply:
This is easy to change but I want to confirm a few things.

1. I think we originally discussed that the reliance on the native RTTI is not permitted, but now we're making it optional, configurable at build time, such that it conditionally replaces our `cetl_rtti`.

2. Our `any` is not a drop-in replacement for `std::any` (unlike, say, `variant` or `span`), so introducing it into the polyfill header as-is seems questionable. The original intention was to introduce a particular instantiation of our `any` into the polyfill namespace, that instantiation being a better approximation of `std::any`. Alternatively, we can avoid polyfilling `std::any` completely considering the differences. Substituting our any template for `std::any` is unlikely to be what we want.

Reply:
With a few tweaks we leave open the possibility that our implementation can be redefined to use RTTI. This seems worth the minor effort to maintain. As for inclusion in the polyfill, let's just leave that for a later time.

Reply:
Understood.

Reply:
No more such header.